python/pylibcudf/pylibcudf/tests/io/test_types.py

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import gc
+import weakref
+
+import pyarrow as pa
+
+import pylibcudf as plc
+
+
+def test_gc_with_table_and_column_input_metadata():
+    class Foo(plc.io.types.TableInputMetadata):
+        def __del__(self):
+            pass
+
+    pa_table = pa.table(
+        {"a": pa.array([1, 2, 3]), "b": pa.array(["a", "b", "c"])}
+    )
+    plc_table = plc.interop.from_arrow(pa_table)
+
+    tbl_meta = Foo(plc_table)
```

Comment:
```suggestion
    tbl_meta = TableInputMetadata(plc_table)
```

Reply:
I got `TypeError: cannot create weak reference to 'pylibcudf.io.types.TableInputMetadata' object`. I think I'd need to implement `__weakref__` in the cython class to get it to work. Similar to `__del__`, its probably not worth it given I'm only using in this test.